uobtheatre/utils/schema.py

```diff
@@ -35,6 +61,192 @@ def mutate(cls, root, info, **inputs):
         return super().mutate(root, info, **inputs)
 
 
+class PermissionNode(graphene.ObjectType):
+    name = graphene.String()
+    description = graphene.String()
+    user_can_assign = graphene.Boolean(default_value=False)
+
+
+class UserPermissionsNode(graphene.ObjectType):
+    user = graphene.Field(ExtendedUserNode)
+    assigned_permissions = graphene.List(graphene.String)
+
+
+class AssignedUsersMixin:
+    """Adds schema objects to show assigned users as well as availble permissions. Authorisation is based on the object level change permission"""
+
+    assigned_users = graphene.List(UserPermissionsNode)
+    assignable_permissions = graphene.List(PermissionNode)
+
+    def resolve_assigned_users(self, info):
+        if not info.context.user.has_perm("change_" + self._meta.model_name, self):
+            return None
+
+        return [
+            UserPermissionsNode(
+                user=user,
+                assigned_permissions=permissions,
+            )
+            for (user, permissions) in get_users_with_perms(
+                self, attach_perms=True, with_group_users=False
+            ).items()
+        ]
+
+    def resolve_assignable_permissions(self, info):
+        if not info.context.user.has_perm("change_" + str(self._meta.model_name), self):
+            return None
+
+        if not isinstance(self, PermissionableModel):  # pragma: no cover
+            return None
+
+        return self.available_permissions_for_user(info.context.user)
+
+
+class SafeFormMutation(SafeMutation, DjangoModelFormMutation):
+    """Provides a wrapper around the DjangoModelFormMutation for use with out validation and authorization system"""
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    # pylint: disable=C0116,arguments-differ
```

Comment:
```suggestion
    # pylint: disable=arguments-differ
```

Lets add `__init_subclass_with_meta__` to the "no-docstring-rgx" in pylintrc